tests/postgres/postgres.rs

```diff
@@ -887,3 +887,14 @@ from (values (null)) vals(val)
 
     Ok(())
 }
+
+#[cfg(feature = "serialize")]
+#[sqlx_macros::test]
+async fn it_row_serializes_to_json() -> anyhow::Result<()> {
+    let mut conn = new::<Postgres>().await?;
+
+    let json = serde_json::to_string(&sqlx::query!(" select (1) as id ").fetch_all(&pool).await?)?;
+    assert_eq!(&json, r#"[{"id":1}]"#);
+
+    Ok(())
+}
```

Comment:
This isn't a sufficient test to ensure that the codegen works for external users. Maybe add a small project in `examples/postgres/` to demonstrate (which will be covered whenever we get around to turning on CI for examples).

Reply:
OK, I'll write do the equivalent of examples/postgres/json unless you have more preferences.

Reply:
Hey @abonander, thanks for clarifying the concern that the feature flag would cause compile time errors for users' existing code.  And also thanks for the pointer to #916.

We might consider using a visually similar syntax for specifying the derives, such as:

`sqlx::query!("SELECT * FROM foo", derive(Serialize))`

Given that the feature flag appears to be unworkable, I'm ready to close this PR in favor of a configurable `query!()` macro.

Reply:
That old conflict with a call to a function named derive, though not sure whether that's a problem 😄

Reply:
Good point. Even though it could be disambiguated by matching derive(($id:ident),*), yea, that would still break any code that might pass `derive(a,b,c...)` as a bound SQL parameter.